src/H5Fpublic.h

```diff
@@ -798,6 +798,9 @@ H5_DLL ssize_t H5Fget_obj_ids(hid_t file_id, unsigned types, size_t max_objs, hi
  *
  */
 H5_DLL herr_t H5Fget_vfd_handle(hid_t file_id, hid_t fapl, void **file_handle);
+
+H5_DLL herr_t H5Fget_file_driver_handle(hid_t file_id, void **file_handle);
+
```

Comment:
If this is not really an API call for the general public, it should probably go in a new H5Fdevelop.h header file.

Reply:
My intention with this API is to create the foundation for a next-generation I/O filter interface: filters would be able to retrieve the VFD handle via this new API and perform memory allocation via the VFD driver (see PR #1500), using functions from the H5FD* family.

Would you have any advice in addition to your comments above to make this API suitable for the general public? Thanks!